user-service/README.md

```diff
@@ -1,26 +1,111 @@
+| UserRequest     | UserResponse     | AuditErrorMessage | AuditableUserResponse |
+|:----------------|:-----------------|:------------------|:----------------------|
+| *token          | *token           | *token            | *token                | 
+| userId          | userId           | userId            | UserResponse          |  
+| resourceId      | resourceId       | resourceId        | AuditErrorMessage     |
+| context         | context          | context           |                       |
+|                 | User             | exception         |                       | 
+|                 |                  | serverMetadata    |                       |
+  
+(fields marked with * are acquired from headers metadata)
 
-The user service separates this concern from the rest of the system. Other components use this service's API to request user details. Some deployments may also allow Palisade to add users to the system, hence the presence of the `addUser()` method in the `UserService` interface.
+The service accepts a `UserRequest` and a token, then checks if the User is in the cache; the technology of which is chosen in the User Services application.yaml 
+and Spring profiles. If the User exists in the cache, then it, alongside any additional attributes, are retrieved and used to create a `UserReponse` object 
+which is then packaged into a `AuditableUserResponse` object with no errors. This is then sent, via Kafka, to the [Resource Service](../resource-service) for further processing. 
+If the User does not exist in the cache then the request gets directed to the local implementation of the User service. If the User service implementation returns a User then a 
+`UserResponse` object gets created and added to an `AuditableUserResponse` object which is sent to the [Resource service](../resource-service).
+Should any errors get thrown when trying to access a User, for example a `NoSuchUserIdException`, then this information creates an `AuditErrorMessage` 
+that is then sent to the [Audit service](../audit-service) to be audited appropriately.
 
-## API Usage
+## REST Interface
 
-Currently the `UserService` supports adding and retrieving of user details via the two following API calls.
+The application exposes two REST endpoints used for debugging or mocking kafka entrypoints:
+* `POST api/user`
+  - accepts an `x-request-token` `String` header, any number of extra headers, and a single `UserRequest` in the body
+  - returns a `202 ACCEPTED` after writing the headers and `AuditableUserResponse` to kafka
+* `POST api/user/multi`
+  - accepts an `x-request-token` `String` header, any number of extra headers, and a list of `UserRequest`s within the body
+  - returns a `202 ACCEPTED` after writing the headers and `AuditableUserResponse` to kafka
```

Comment:
It writes a UserRequest to kafka and returns, no?

Reply:
You're quite right. Updated so that a UserRequest is sent to Kafka :D